src/Language/PureScript/Make.hs

```diff
@@ -238,7 +250,9 @@ make ma@MakeActions{..} ms = do
                 _ -> return e
             foldM go env deps
           env <- C.readMVar (bpEnv buildPlan)
-          (exts, warnings) <- listen $ rebuildModule' ma env externs m
+          idx <- C.takeMVar (bpIndex buildPlan)
+          (exts, warnings) <- listen $ rebuildModuleWithIndex ma env externs m (Just (idx, cnt))
+          C.putMVar (bpIndex buildPlan) (idx + 1)
```

Comment:
I'm not intimately familiar with GHC's concurrency primitives, but would `takeMVar` and `putMVar` make it so `rebuildModuleWithIndex` is not performed concurrently?